### DISCUSSION TITLE
Loading an image with multiple redirects

### DISCUSSION THREAD

**Discussion by Sam-Spencer** [kean/NukeUI#27] [Q&A]
Hi Nuke community!

I'm working on integrating Nuke into an iOS app and came across an issue with loading an image from an API call using multiple redirects. Just curious if Nuke supports this, or if I need to build my own `ImagePipeline` to handle such scenarios.

Here's an example of what the API responds with:

 1.  Initial request and response
     ```
      curl 'https://myImageAPI.com/api/image?width=1920
      HTTP/2 301 
     ```
 2. The API uses a 301 redirect to either deliver the image or perform an additional redirect to a fallback image if it is unavailable.
     ```
      found image response: HTTP2/ 200, includes image data in body
      no image response: HTTP2/ 307, moves on to next step 
     ```
 3. If the image is unavailable, the API finally responds with a fallback image:
     ```
     HTTP2/ 200, includes image data in body
     ```

It seems that Nuke doesn't handle the subsequent redirects from the API and only reports success if image data is returned in the first request. Is this supported by Nuke? If so, does anyone have an example on how to handle this? If not, what's the recommended way to go about handling such a situation? Building a custom `ImagePipeline` publisher?

**Comment by kean** [comment]
Hey, @Sam-Spencer. Have you figured it out? If yes, could you please post an update?

I haven't worked with redirects, but unless they are supported by default by `URLSession` / `DataLoader`, it needs to be configured to support them. I added a ticket to the board https://trello.com/c/JI4alJCi/20-improve-subscribing-to-existing-processing-tasks
